packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -73,30 +73,48 @@ class ErrorHandlingFileSystem extends ForwardingFileSystem {
   /// This method should be preferred to checking if it exists and
   /// then deleting, because it handles the edge case where the file or directory
   /// is deleted by a different program between the two calls.
-  static bool deleteIfExists(FileSystemEntity file, {bool recursive = false}) {
+  static bool deleteIfExists(FileSystemEntity file, {bool recursive = false, bool handleSharingViolation = false}) {
     if (!file.existsSync()) {
       return false;
     }
     try {
       file.deleteSync(recursive: recursive);
     } on FileSystemException catch (err) {
+      final int errorCode = err?.osError?.errorCode;
       // Certain error codes indicate the file could not be found. It could have
       // been deleted by a different program while the tool was running.
       // if it still exists, the file likely exists on a read-only volume.
       //
       // On windows this is error code 2: ERROR_FILE_NOT_FOUND, and on
       // macOS/Linux it is error code 2/ENOENT: No such file or directory.
       const int kSystemCannotFindFile = 2;
-      if (err?.osError?.errorCode != kSystemCannotFindFile || _noExitOnFailure) {
+      // Sharing violation indicates that another program has this file
+      // open. This is only an error on windows.
+      const int kSharingViolation = 32;
+
+      if (_noExitOnFailure) {
         rethrow;
       }
-      if (file.existsSync()) {
+
+      if (handleSharingViolation && errorCode == kSharingViolation) {
```

Comment:
Check that the platform is Windows. 32 is something else on e.g. Linux.

Reply:
Since this is a static method, i don't have the platform. In this case, I only set handleSharingViolation to true when platform is windows. I could change the API so that platform is always passed in and do the logic here instead of the cache. WDUT?

Reply:
I'm sorry for the churn, but I'm beginning to wonder if my original suggestion to move the logic here was the right thing to do. I do think we need to rethink this method along the lines of https://github.com/flutter/flutter/pull/75719#discussion_r573250670, but if the aim is to cherry-pick a fix for the prominent crasher, then handling this error at the problematic call site might actually give us something more maintainable.

Reply:
Yeah, SGTM